### PR TITLE
feat(encounter): scanner screen

### PR DESCRIPTION
**Pull request by digorath**
## Changes proposed in this PR:

- replaced expo barcode scanner with expo camera
  - barcode scanner was not be able to properly resized for android (fixed aspect ratio)
- added new screen for scanning encouter qr codes
- added validation for encounter user type
- added dummy hooks for creating an encounter

---

SVA-287

### CODE REVIEW

File: src/hooks/encounter/encounter.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  loading: boolean;
+} => {
+  const [loading, setLoading] = useState(false);
+
+  const createEncounter = useCallback(async (qrId: string, userId: string) => {
+    setLoading(true);
+    try {
+      const res = await createEncounterAsync(qrId, userId);
+
+      if (!res.data || res.error) {
```
Comment (donni106) [inline src/hooks/encounter/encounter.ts:45]: i think it will be necessary to check for the status code to decide wether there was success or an error

Comment (digorath) [reply src/hooks/encounter/encounter.ts:45]: i will revisit all error handling upon implementing the api calls.

i don't know whether i will include the error logic here, or already inside of the `createEncounterAsync` function.

Comment (digorath) [reply src/hooks/encounter/encounter.ts:45]: changed in #339 
`createEncounterAsync` will now return a parsed and validated result or nothing. so we only need to check for presence of `res` here. (the used data is no longer nested as `data` but is directly returned)

File: src/screens/EncounterScannerScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import { useCreateEncounter, useEncounterUser } from '../hooks';
+import { User } from '../types';
+
+const showErrorAlert = () =>
+  Alert.alert(texts.encounter.errorScanTitle, texts.encounter.errorScanBody);
+
+const parseQrCode = (data: string): string | undefined => {
+  const result = Linking.parse(data);
+
+  if (result.scheme === (__DEV__ ? 'exp' : appJson.expo.scheme) && result.path === 'encounter') {
```
Comment (donni106) [inline src/screens/EncounterScannerScreen.tsx:27]: i miss an explanation for `exp`. why this in dev mode? i have a clue but cannot be sure.

File: src/screens/EncounterScannerScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+          <BoldText>{texts.encounter.scannerSubTitle}</BoldText>
+        </Wrapper>
+      </WrapperWithOrientation>
+      {isScanning ? (
+        <Wrapper style={styles.scannerContainer}>
+          <Camera
+            ratio="1:1"
+            onBarCodeScanned={handleBarCodeScanned}
+            style={styles.scanner}
+            // barCodeTypes={['org.iso.QRCode']}
```
Comment (donni106) [inline src/screens/EncounterScannerScreen.tsx:114]: comment needed for later?

Comment (digorath) [reply src/screens/EncounterScannerScreen.tsx]: nope, that is a comment that is a remainder of a previous version.
this prop is not supported here.

i removed it now

File: src/screens/EncounterScannerScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    [navigation]
+  );
+
+  const { loading: loadingCreateEncounter, createEncounter } = useCreateEncounter(
+    onCreationSuccess,
+    showErrorAlert
+  );
+
+  const handleBarCodeScanned = ({ data, type }: BarCodeScanningResult) => {
+    if (type !== (device.platform === 'android' ? 256 : 'org.iso.QRCode')) {
```
[the remark this answers is not in the document]
Comment (digorath) [reply src/screens/EncounterScannerScreen.tsx:56]: i got this from scanning the qr code, that we ourselves created.

this way we do not react to codes of a different type, if one shows up in the camera view

File: src/screens/EncounterScannerScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+export const EncounterScannerScreen = ({ navigation }: { navigation: any }) => {
+  const { loading: loadingUser, userId } = useEncounterUser();
+  const [isScanning, setIsScanning] = useState(true);
+  const [hasPermission, setHasPermission] = useState<boolean>();
+
+  const onCreationSuccess = useCallback(
+    (data: User) => {
+      // TODO: add navigation here instead of console.log
+      // navigation.navigate(ScreenName.EncounterUserDetails, { data });
+      console.log(data);
```
[the remark this answers is not in the document]
Comment (digorath) [reply src/screens/EncounterScannerScreen.tsx:45]: ✔️

File: src/screens/EncounterScannerScreen.tsx (hunk 2 above)
Comment (digorath) [reply src/screens/EncounterScannerScreen.tsx:29]: i added a comment for explanation

**Pull request merged**